### PR TITLE
triplestore class which does add, delete/update and get mappings/triples

### PR DESCRIPTION
**Pull request by Treesarj**

### CODE REVIEW

File: oteapi/triplestore/triplestore.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        Returns:
+            The output of the search query in the form of subject, object and predicate.
+
+        """
+        connection = self.server.openSession(
+            reason("<" + str(self.config.repositoryName) + ">")
+        )
+        tuple_query = connection.prepareTupleQuery(query=sparql_query)
+        response = []
+        results = tuple_query.evaluate(output_format=TupleFormat.JSON)
```
Comment (kriwiik) [inline oteapi/triplestore/triplestore.py:100]: I believe this line can be removed.

File: oteapi/triplestore/triplestore.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                    triple_set["s"] = str(result.getValue("s"))
+                if "'p': " in str(result):
+                    triple_set["p"] = str(result.getValue("p"))
+                if "'o': " in str(result):
+                    triple_set["o"] = str(result.getValue("o"))
+                response.append((triple_set))
+        connection.close()
+        return json.dumps(response)
+
+    def update_delete(self, sparql_query: str) -> None:
```
Comment (CasperWA) [inline oteapi/triplestore/triplestore.py:113]:
```suggestion
    def update(self, sparql_query: str) -> None:
```
I think `update` is an appropriate name here - isn't this really an overwrite? Or does it truly delete some triples as well completely?

File: requirements.txt [hunk 3]
```diff
@@ -1,3 +1,4 @@
+agraph-python~=101.0.7
```
Comment (CasperWA) [inline requirements.txt:1]:
```suggestion
agraph-python>=101.0.7,<102
```

**Review by CasperWA — DISMISSED** [review]

File: oteapi/README.md [hunk 4]
```diff
@@ -5,3 +5,4 @@
 * models - Generic configuration data models used by the strategies.
 * plugins - Module to implement the plugin and factory functionalities.
 * strategies - Implementation of the core strategies.
+* triplestore - A triplestore to share mappings between plugins.
```
[the remark this answers is not in the document]
Comment (kriwiik) [reply oteapi/README.md:8]: I think triplestore and triple store (no hyphen) are the most commonly used spellings

[the remark this answers is not in the document]
Comment (CasperWA) [reply oteapi/README.md:8]: https://en.wikipedia.org/wiki/Triplestore Agreed

File: oteapi/models/triplestoreconfig.py [hunk 5]
```diff
@@ -13,17 +13,22 @@ class TripleStoreConfig(AttrDict):
     a configuration object.
     """
 
-    repositoryName: str = Field(...,
-        description="repository name where the mappings need to be stored.")
-    agraphHost: str = Field(...,
+    repositoryName: str = Field(
```
[the remark this answers is not in the document]
Comment (CasperWA) [reply oteapi/models/triplestoreconfig.py:16]: The `TripleStoreConfig` is dissimilar to all the other `*Config` classes, which represent general strategy configurations.

Rather, `TripleStoreConfig` is similar to the `DataCacheConfig`, upon which it has also been modellen (I believe) :)

File: oteapi/triplestore/triplestore.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        response = []
+        with tuple_query.evaluate(output_format=TupleFormat.JSON) as results:
+            for result in results:
+                triple_set = {}
+                if "'s': " in str(result):
+                    triple_set["s"] = str(result.getValue("s"))
+                if "'p': " in str(result):
+                    triple_set["p"] = str(result.getValue("p"))
+                if "'o': " in str(result):
+                    triple_set["o"] = str(result.getValue("o"))
```
[the remark this answers is not in the document]
Comment (Treesarj) [reply oteapi/triplestore/triplestore.py]: It will return an empty list

**Comment by jesper-friis** [comment]
Looks very nice. A unit test would be nice to have, but that could also be added in a later PR.

**Comment by jesper-friis** [comment]
Looks very nice. A unit test would be nice to have, but that could also be added in a later PR.

**Comment by Treesarj** [comment]
> Looks very nice. A unit test would be nice to have, but that could also be added in a later PR.

I checked to add a test but it requires a running instance of AllegroGraph. I couldn't find a away ti mock it

**Review by kriwiik — APPROVED** [review]

**Comment by Treesarj** [comment]
dismissing @CasperWA 's review as it's blocking the merge! as discussed I have added the requested changes, but some got outdated due to some line changes

**Pull request merged**